app/src/main/java/com/example/Egida/domain/useCase/UserUSeCaseImpl.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.Egida.domain.useCase
+
+import com.example.Egida.data.Database
+import com.example.Egida.domain.entity.User
+
+class UserUSeCaseImpl(
+    private val database: Database
+) : UserUseCase {
+
+    override fun checkingUser(): Boolean {
```

Comment:
checkUser()

Reply:
Also, it's more reasonable to return sealed class as a result of this check. It could be more, than 2 variants: user exists, user not exists, token required, email checking. And for each state you may want to do any specific actions.